src/adapter/src/catalog/builtin.rs

```diff
@@ -2549,6 +2549,32 @@ FROM
         JOIN mz_internal.mz_cluster_replica_metrics AS m ON m.replica_id = r.id",
 };
 
+pub const MZ_DATAFLOW_OPERATOR_PARENTS: BuiltinView = BuiltinView {
+    name: "mz_dataflow_operator_parents",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE VIEW mz_internal.mz_operator_parents AS
+WITH
+	parent_addrs
+		AS (
+			SELECT
+				id,
+				address[1:list_length(address) - 1]
+					AS parent_address,
+				worker_id
+			FROM
+				mz_internal.mz_dataflow_addresses AS mda
+				INNER JOIN mz_internal.mz_dataflow_operators
+						AS mdo USING (id, worker_id)
+		)
+SELECT
+	pa.id, mda.id AS parent_id
+FROM
+	parent_addrs AS pa
+	JOIN mz_internal.mz_dataflow_addresses AS mda ON
+			pa.parent_address = mda.address
+			AND pa.worker_id = mda.worker_id",
```

Comment:
I find that formatting hard to read. How about:

```suggestion
WITH parent_addrs AS (
    SELECT
        id,
        address[1:list_length(address) - 1] AS parent_address,
        worker_id
    FROM mz_internal.mz_dataflow_addresses
        INNER JOIN mz_internal.mz_dataflow_operators
            USING (id, worker_id)
)
SELECT pa.id, mda.id AS parent_id
FROM parent_addrs AS pa
    JOIN mz_internal.mz_dataflow_addresses AS mda
        ON pa.parent_address = mda.address
        AND pa.worker_id = mda.worker_id",
```